sageintacctsdk/apis/api_base.py

```diff
@@ -196,6 +201,12 @@ def __post_request(self, dict_body: dict, api_url: str):
         parsed_response = json.loads(json.dumps(parsed_xml))
 
         if raw_response.status_code == 200:
+            if (parsed_response['response']['control']['status'] == 'failure' or parsed_response['response']['operation']['authentication']['status'] == 'failure' or \
```

Comment:
handle all keys safely, they may not exist in all case. do it like `parsed_response.get('response', {}).get('control', {}).get('status', {})`